src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int ID_INDEX = 1;
+    private static final int ID_POSITION = 3;
+    private static final int PRICE_POSITION = 2;
+    private static final int TITLE_POSITION = 1;
```

Comment:
I consider it unnecessary since when you set some parameter, you set it sequentially -- 1, 2, etc. It's just ordering, not some important index of data to parse or something like that.